<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="A collection of functions for visualizing,exploring and annotating genetic association results.Association results from multiple traits can be viewed simultaneously along with gene annotation, over the entire genome (Manhattan plot) or in the more detailed regional view.">
<title>Create Custom Plots for Viewing Genetic Association Results • topr</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Create Custom Plots for Viewing Genetic Association Results">
<meta property="og:description" content="A collection of functions for visualizing,exploring and annotating genetic association results.Association results from multiple traits can be viewed simultaneously along with gene annotation, over the entire genome (Manhattan plot) or in the more detailed regional view.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">topr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/topr.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/manhattan.html">Intro into manhattan</a>
    <a class="dropdown-item" href="articles/regionplot.html">Intro into regionplot</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/totajuliusd/topr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><p>Note!!! <em>topr</em> v.2.0.0 can be used with any species and any number of chromosomes!! <br></p>
<p>See <a href="https://github.com/totajuliusd/topr?tab=readme-ov-file#how-to-use-topr-with-other-species-than-human" class="external-link uri">https://github.com/totajuliusd/topr?tab=readme-ov-file#how-to-use-topr-with-other-species-than-human</a></p>
<div class="section level1">
<div class="page-header"><h1 id="topr-an-r-package-for-viewing-and-annotating-genetic-association-results">
<em>topr</em>: an R package for viewing and annotating genetic association results<a class="anchor" aria-label="anchor" href="#topr-an-r-package-for-viewing-and-annotating-genetic-association-results"></a>
</h1></div>
<p><img src="reference/figures/manhattan_wide_1080_high.gif" alt="topr GIF" width="100%"></p>
<details><summary><i>Click here to see the commands used in the .gif above</i>
</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/totajuliusd/topr" class="external-link">topr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Single GWAS manhattan plots</span></span>
<span><span class="co"># Start by taking a look at one of topr's inbuilt datasets</span></span>
<span><span class="va">CD_UKBB</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="va">CD_UKBB</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, annotate<span class="op">=</span><span class="fl">5e-9</span><span class="op">)</span></span>
<span><span class="va">CD_UKBB</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="reference/get_lead_snps.html">get_lead_snps</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">CD_UKBB</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="reference/get_lead_snps.html">get_lead_snps</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="reference/annotate_with_nearest_gene.html">annotate_with_nearest_gene</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, annotate<span class="op">=</span><span class="fl">1e-9</span>, highlight_genes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FTO"</span>,<span class="st">"THADA"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, annotate<span class="op">=</span><span class="fl">1e-9</span>, highlight_genes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FTO"</span>,<span class="st">"THADA"</span><span class="op">)</span>, color<span class="op">=</span><span class="st">"darkred"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Multi GWAS manhattan/miami plots</span></span>
<span><span class="va">CD_FINNGEN</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span><span class="op">)</span>, legend_labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UKBB"</span>,<span class="st">"FINNGEN"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span><span class="op">)</span>, legend_labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UKBB"</span>,<span class="st">"FINNGEN"</span><span class="op">)</span>, ntop<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Add the third GWAS result... </span></span>
<span><span class="va">UC_UKBB</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span>, <span class="va">UC_UKBB</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span>, <span class="va">UC_UKBB</span><span class="op">)</span>, legend_labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD UKBB"</span>,<span class="st">"CD FINNGEN"</span>, <span class="st">"UC UKBB"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># display the first input dataset on the top plot (ntop = 1)</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span>, <span class="va">UC_UKBB</span><span class="op">)</span>, legend_labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD UKBB"</span>,<span class="st">"CD FINNGEN"</span>, <span class="st">"UC UKBB"</span><span class="op">)</span>, ntop<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="co"># display the first TWO input datasets on the top plot (ntop = 2)</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span>, <span class="va">UC_UKBB</span><span class="op">)</span>, legend_labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD UKBB"</span>,<span class="st">"CD FINNGEN"</span>, <span class="st">"UC UKBB"</span><span class="op">)</span>, ntop<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span>, <span class="va">UC_UKBB</span><span class="op">)</span>, legend_labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD UKBB"</span>,<span class="st">"CD FINNGEN"</span>, <span class="st">"UC UKBB"</span><span class="op">)</span>, ntop<span class="op">=</span><span class="fl">2</span>, annotate<span class="op">=</span><span class="fl">1e-12</span><span class="op">)</span></span>
<span><span class="co">#apply different annotation thresholds to the three datasets ( annotate = c(5e-9,1e-12,1e-15) )</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span>, <span class="va">UC_UKBB</span><span class="op">)</span>, legend_labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD UKBB"</span>,<span class="st">"CD FINNGEN"</span>, <span class="st">"UC UKBB"</span><span class="op">)</span>, ntop<span class="op">=</span><span class="fl">2</span>, annotate<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5e-9</span>,<span class="fl">1e-12</span>,<span class="fl">1e-15</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Make the plot look prettier by resetting they scales of the x and y  axes, changing the angle of the annotation text (angle = 90), moving it up a bit (nudge_y = 12) and slightly reducing the size of all text (scale = 0.7)</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span>, <span class="va">UC_UKBB</span><span class="op">)</span>, legend_labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD UKBB"</span>,<span class="st">"CD FINNGEN"</span>, <span class="st">"UC UKBB"</span><span class="op">)</span>, ntop<span class="op">=</span><span class="fl">2</span>, annotate<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5e-9</span>,<span class="fl">1e-12</span>,<span class="fl">1e-15</span><span class="op">)</span>, ymax<span class="op">=</span><span class="fl">65</span>, ymin<span class="op">=</span><span class="op">-</span><span class="fl">55</span>, nudge_y<span class="op">=</span><span class="fl">12</span>,angle<span class="op">=</span><span class="fl">90</span>, scale<span class="op">=</span><span class="fl">0.7</span><span class="op">)</span></span>
<span><span class="co"># The same plot as above in the old topr grey theme (theme_grey = T)</span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span>, <span class="va">UC_UKBB</span><span class="op">)</span>, legend_labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD UKBB"</span>,<span class="st">"CD FINNGEN"</span>, <span class="st">"UC UKBB"</span><span class="op">)</span>, ntop<span class="op">=</span><span class="fl">2</span>, annotate<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5e-9</span>,<span class="fl">1e-12</span>,<span class="fl">1e-15</span><span class="op">)</span>, ymax<span class="op">=</span><span class="fl">65</span>, ymin<span class="op">=</span><span class="op">-</span><span class="fl">55</span>, nudge_y<span class="op">=</span><span class="fl">12</span>,angle<span class="op">=</span><span class="fl">90</span>, scale<span class="op">=</span><span class="fl">0.7</span>, theme_grey <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
</details><p>For more examples see the <a href="https://totajuliusd.github.io/topr_manual/" class="external-link">topr webpage</a>.</p>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Please cite the following paper if you use <em>topr</em> in a publication:</p>
<p>Juliusdottir, T. <em>topr</em>: an R package for viewing and annotating genetic association results. BMC Bioinformatics 24, 268 (2023). <a href="https://doi.org/10.1186/s12859-023-05301-4" class="external-link uri">https://doi.org/10.1186/s12859-023-05301-4</a></p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<hr>
<p>Install from CRAN:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"topr"</span><span class="op">)</span></span></code></pre></div>
<p>Or from github:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"totajuliusd/topr"</span><span class="op">)</span></span></code></pre></div>
<p>And then load the package:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/totajuliusd/topr" class="external-link">topr</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="main-features-and-functionality">Main features and functionality<a class="anchor" aria-label="anchor" href="#main-features-and-functionality"></a>
</h2>
<hr>
<p><em>topr</em> is written in the R programming language and utilises the ggplot2 and ggrepel R graphics libraries for plotting.</p>
<p><em>topr’s</em> two main plot functions are <code><a href="reference/manhattan.html">manhattan()</a></code> and <code><a href="reference/regionplot.html">regionplot()</a></code>. See the <a href="https://totajuliusd.github.io/topr_manual/" class="external-link">topr webpage</a> for more details on all of <em>topr’s</em> plot functions.</p>
<p>The manhattan() function returns a <strong>ggplot object</strong>. The regionplot() function draws three ggplotGrobs aligned using egg::gtable_frame, however it can be called with the <code>extract_plots</code> argument set to TRUE to return a list of three <strong>ggplot objects</strong> instead.</p>
<div class="section level3">
<h3 id="example-input-datasets">Example input datasets<a class="anchor" aria-label="anchor" href="#example-input-datasets"></a>
</h3>
<p><em>See <a href="https://totajuliusd.github.io/topr/articles/input_datasets.html" class="external-link">Input datasets</a> for more detailed information.</em></p>
<p>Input datasets must include least three columns (<code>CHROM, POS</code> and <code>P</code>), where naming of the columns is flexible (i.e the chr label can be either chr or chrom and is case insensitive).</p>
<p><em>topr</em> has 3 inbuilt GWAS results (<code>CD_UKBB, CD_FINNGEN</code> and <code>UC_UKBB</code>). To get information on them, do:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">?</span><span class="va">CD_UKBB</span></span>
<span><span class="op">?</span><span class="va">CD_FINNGEN</span></span>
<span><span class="op">?</span><span class="va">UC_UKBB</span></span></code></pre></div>
<p>The chromosome in the <code>CHROM</code> column can be represented with or without the <i>chr</i> suffix, e.g (chr1 or 1)</p>
</div>
<div class="section level3">
<h3 id="basic-usage">Basic usage<a class="anchor" aria-label="anchor" href="#basic-usage"></a>
</h3>
<hr>
<p>Basic usage of <em>topr’s</em> key functions is as follows. <br></p>
<div class="section level4">
<h4 id="manhattan">Manhattan<a class="anchor" aria-label="anchor" href="#manhattan"></a>
</h4>
<hr>
<p>See <a href="https://totajuliusd.github.io/topr_manual/manhattan.html" class="external-link">manhattan</a> for more detailed examples of how to use the manhattan plot function.</p>
<p>View the whole genome association results on a Manhattan plot:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="va">CD_UKBB</span><span class="op">)</span></span></code></pre></div>
<p>Annotate the lead/index variants (with p-values below 5e-9) with their nearest gene:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, annotate<span class="op">=</span><span class="fl">5e-9</span><span class="op">)</span></span></code></pre></div>
<p>View multiple GWAS results on the same plot</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">manhattan</span>(<span class="fu">list</span>(CD_UKBB, CD_FINNGEN), <span class="at">legend_labels =</span> <span class="fu">c</span>(<span class="st">"UKBB"</span>, FinnGen<span class="st">"))</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level4">
<h4 id="regionplot">Regionplot<a class="anchor" aria-label="anchor" href="#regionplot"></a>
</h4>
<hr>
<p>See <a href="https://totajuliusd.github.io/topr_manual/regionplot.html" class="external-link">regionplot</a> for more detailed examples of how to use the regionplot function.</p>
<p>Further zoom-in on a genetic region by gene name (<em>IL23R</em>):</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/regionplot.html">regionplot</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, gene<span class="op">=</span><span class="st">"IL23R"</span><span class="op">)</span></span></code></pre></div>
<p>View the correlation pattern between the variants within the region in a locuszoom like plot. Note that the variant correlation (<code>R2</code>) has to be pre-calculated and included in the input dataframe.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/locuszoom.html">locuszoom</a></span><span class="op">(</span><span class="va">R2_CD_UKBB</span><span class="op">)</span></span></code></pre></div>
<p>Display multiple GWAS results zoomed in on the <em>IL23R</em> gene</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/regionplot.html">regionplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">UC_UKBB</span>, <span class="va">CD_UKBB</span><span class="op">)</span>, gene<span class="op">=</span><span class="st">"IL23R"</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level4">
<h4 id="other-useful-functions">Other useful functions<a class="anchor" aria-label="anchor" href="#other-useful-functions"></a>
</h4>
<hr>
<p>Extract lead/index variants from the GWAS dataset (<code>CD_UKBB</code>):</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_lead_snps.html">get_lead_snps</a></span><span class="op">(</span><span class="va">CD_UKBB</span><span class="op">)</span></span></code></pre></div>
<p>Annotate the lead/index variants with their nearest gene:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_lead_snps.html">get_lead_snps</a></span><span class="op">(</span><span class="va">CD_UKBB</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="reference/annotate_with_nearest_gene.html">annotate_with_nearest_gene</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Get genomic coordinates for a gene (<em>topr</em> uses genome build GRCh38.p13 by default):</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_gene_coords.html">get_gene_coords</a></span><span class="op">(</span><span class="st">"IL23R"</span><span class="op">)</span></span></code></pre></div>
<p>Get genomic coordinates for a gene using genome build GRCh37 instead.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_gene_coords.html">get_gene_coords</a></span><span class="op">(</span><span class="st">"IL23R"</span>, build<span class="op">=</span><span class="st">"37"</span><span class="op">)</span></span></code></pre></div>
<p>Get snps within a region:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_snps_within_region.html">get_snps_within_region</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, region <span class="op">=</span> <span class="st">"chr1:67138906-67259979"</span><span class="op">)</span></span></code></pre></div>
<p>Get the top variant on a chromosome:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_top_snp.html">get_top_snp</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, chr<span class="op">=</span><span class="st">"chr1"</span><span class="op">)</span></span></code></pre></div>
<p>Create a snpset by extracting the top/lead variants from the CD_UKBB dataset and overlapping variants (by position) in the CD_FINNGEN dataset.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_snpset.html">get_snpset</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span><span class="op">)</span></span></code></pre></div>
<p>Create an effecplot by plotting the effect sizes of top/lead variants overlapping in two datasets.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/effectplot.html">effectplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span><span class="op">)</span>, annotate <span class="op">=</span> <span class="fl">1e-08</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level4">
<h4 id="how-to-use-topr-with-other-species-than-human">How to use <em>topr</em> with other species than human<a class="anchor" aria-label="anchor" href="#how-to-use-topr-with-other-species-than-human"></a>
</h4>
<hr>
<p>By default <em>topr</em> uses the human genome assembly for annotation. As of from version 2.0.0. topr can be used with different gene annotations as long as they are provided by the user in a specific format.</p>
<p>Required columns in the gene annotation file are the following: <code>chrom,gene_start_gene_end,gene_symbol,biotype,exon_chromstart</code> and <code>exon_chromend</code></p>
<p>To see an example of a gene annotation file, view the inbuilt human genome annotation file (hg38) from the <code>enshuman</code> package which comes with topr v.2.0.0:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span>n<span class="op">=</span><span class="fl">2</span>, <span class="fu">enshuman</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/enshuman/man/hg38.html" class="external-link">hg38</a></span><span class="op">)</span></span></code></pre></div>
<p>Note that there is no <em>“chr”</em> prefix in front of the chromosome name. <br></p>
<div class="section level5">
<h5 id="example-using-the-mouse-genome">Example using the mouse genome<a class="anchor" aria-label="anchor" href="#example-using-the-mouse-genome"></a>
</h5>
<p>Start by downloading and unzipping the <strong>mouse gtf</strong> file. From within the terminal this can be done as follows:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>wget https<span class="sc">:</span><span class="er">//</span>ftp.ensembl.org<span class="sc">/</span>pub<span class="sc">/</span>current<span class="sc">/</span>gtf<span class="sc">/</span>mus_musculus<span class="sc">/</span>Mus_musculus.GRCm39.<span class="fl">111.</span>gtf.gz</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>gunzip Mus_musculus.GRCm39.<span class="fl">111.</span>gtf.gz</span></code></pre></div>
<p>Next convert the file into the format required by topr. This can for example be done using the <code>mk_annotation_file.py</code> script (the content of the script is shown below the command).</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>python mk_annotation_file.py Mus_musculus.GRCm39.<span class="fl">111.</span>gtf <span class="sc">&gt;</span> Mus_musculus.GRCm39.<span class="fl">111.</span>tsv</span></code></pre></div>
<br><details><summary><i>Click here to see what the mk_annotation_file.py looks like</i>
</summary><div class="sourceCode" id="cb23" eval="F"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="co">#!/usr/bin/env python                                                                                                                                                                    </span></span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>import sys</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a>import re</span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a><span class="cf">if</span> <span class="fu">len</span>(sys.argv) <span class="sc">!=</span> <span class="dv">2</span><span class="sc">:</span></span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"Usage: {} &lt;file&gt;"</span><span class="fu">.format</span>(sys.argv[<span class="dv">0</span>]))</span>
<span id="cb23-8"><a href="#cb23-8" tabindex="-1"></a>    <span class="fu">sys.exit</span>(<span class="dv">1</span>)</span>
<span id="cb23-9"><a href="#cb23-9" tabindex="-1"></a></span>
<span id="cb23-10"><a href="#cb23-10" tabindex="-1"></a>file_name <span class="ot">=</span> sys.argv[<span class="dv">1</span>]</span>
<span id="cb23-11"><a href="#cb23-11" tabindex="-1"></a></span>
<span id="cb23-12"><a href="#cb23-12" tabindex="-1"></a>try<span class="sc">:</span></span>
<span id="cb23-13"><a href="#cb23-13" tabindex="-1"></a>    with <span class="fu">open</span>(file_name, <span class="st">'r'</span>) as fh<span class="sc">:</span></span>
<span id="cb23-14"><a href="#cb23-14" tabindex="-1"></a>        genes <span class="ot">=</span> {}</span>
<span id="cb23-15"><a href="#cb23-15" tabindex="-1"></a>        <span class="cf">for</span> line <span class="cf">in</span> fh<span class="sc">:</span></span>
<span id="cb23-16"><a href="#cb23-16" tabindex="-1"></a>            <span class="cf">if</span> not <span class="fu">line.startswith</span>(<span class="st">'#'</span>)<span class="sc">:</span></span>
<span id="cb23-17"><a href="#cb23-17" tabindex="-1"></a>                columns <span class="ot">=</span> <span class="fu">line.strip</span>()<span class="fu">.split</span>(<span class="st">'</span><span class="sc">\t</span><span class="st">'</span>)</span>
<span id="cb23-18"><a href="#cb23-18" tabindex="-1"></a>                chrom, start, end <span class="ot">=</span> columns[<span class="dv">0</span>], columns[<span class="dv">3</span>], columns[<span class="dv">4</span>]</span>
<span id="cb23-19"><a href="#cb23-19" tabindex="-1"></a></span>
<span id="cb23-20"><a href="#cb23-20" tabindex="-1"></a>                <span class="cf">if</span> columns[<span class="dv">2</span>] <span class="sc">==</span> <span class="st">'exon'</span><span class="sc">:</span></span>
<span id="cb23-21"><a href="#cb23-21" tabindex="-1"></a>                    gene_name <span class="ot">=</span> None</span>
<span id="cb23-22"><a href="#cb23-22" tabindex="-1"></a>                    match <span class="ot">=</span> <span class="fu">re.search</span>(r<span class="st">'gene_name\s+"(\S+)"'</span>, columns[<span class="dv">8</span>])</span>
<span id="cb23-23"><a href="#cb23-23" tabindex="-1"></a>                    <span class="cf">if</span> match<span class="sc">:</span></span>
<span id="cb23-24"><a href="#cb23-24" tabindex="-1"></a>                        gene_name <span class="ot">=</span> <span class="fu">match.group</span>(<span class="dv">1</span>)</span>
<span id="cb23-25"><a href="#cb23-25" tabindex="-1"></a></span>
<span id="cb23-26"><a href="#cb23-26" tabindex="-1"></a>                    <span class="cf">if</span> gene_name <span class="cf">in</span> genes<span class="sc">:</span></span>
<span id="cb23-27"><a href="#cb23-27" tabindex="-1"></a>                        <span class="cf">if</span> <span class="st">"exon_chromstart"</span> <span class="cf">in</span> genes[gene_name]<span class="sc">:</span></span>
<span id="cb23-28"><a href="#cb23-28" tabindex="-1"></a>                            genes[gene_name][<span class="st">"exon_chromstart"</span>]<span class="fu">.append</span>(start)</span>
<span id="cb23-29"><a href="#cb23-29" tabindex="-1"></a>                        <span class="cf">else</span><span class="sc">:</span></span>
<span id="cb23-30"><a href="#cb23-30" tabindex="-1"></a>                            genes[gene_name][<span class="st">"exon_chromstart"</span>] <span class="ot">=</span> [start]</span>
<span id="cb23-31"><a href="#cb23-31" tabindex="-1"></a></span>
<span id="cb23-32"><a href="#cb23-32" tabindex="-1"></a>                        <span class="cf">if</span> <span class="st">"exon_chromend"</span> <span class="cf">in</span> genes[gene_name]<span class="sc">:</span></span>
<span id="cb23-33"><a href="#cb23-33" tabindex="-1"></a>                            genes[gene_name][<span class="st">"exon_chromend"</span>]<span class="fu">.append</span>(end)</span>
<span id="cb23-34"><a href="#cb23-34" tabindex="-1"></a>                        <span class="cf">else</span><span class="sc">:</span></span>
<span id="cb23-35"><a href="#cb23-35" tabindex="-1"></a>                            genes[gene_name][<span class="st">"exon_chromend"</span>] <span class="ot">=</span> [end]</span>
<span id="cb23-36"><a href="#cb23-36" tabindex="-1"></a></span>
<span id="cb23-37"><a href="#cb23-37" tabindex="-1"></a>                elif columns[<span class="dv">2</span>] <span class="sc">==</span> <span class="st">'gene'</span><span class="sc">:</span></span>
<span id="cb23-38"><a href="#cb23-38" tabindex="-1"></a>                    gene_name, gene_type <span class="ot">=</span> None, None</span>
<span id="cb23-39"><a href="#cb23-39" tabindex="-1"></a>                    match <span class="ot">=</span> <span class="fu">re.search</span>(r<span class="st">'gene_name\s+"(\S+)".+gene_biotype\s+"(\S+)"'</span>, columns[<span class="dv">8</span>])</span>
<span id="cb23-40"><a href="#cb23-40" tabindex="-1"></a>                    <span class="cf">if</span> match<span class="sc">:</span></span>
<span id="cb23-41"><a href="#cb23-41" tabindex="-1"></a>                        gene_name, gene_type <span class="ot">=</span> <span class="fu">match.group</span>(<span class="dv">1</span>), <span class="fu">match.group</span>(<span class="dv">2</span>)</span>
<span id="cb23-42"><a href="#cb23-42" tabindex="-1"></a></span>
<span id="cb23-43"><a href="#cb23-43" tabindex="-1"></a>                    <span class="cf">if</span> gene_name not <span class="cf">in</span> genes<span class="sc">:</span></span>
<span id="cb23-44"><a href="#cb23-44" tabindex="-1"></a>                        genes[gene_name] <span class="ot">=</span> {</span>
<span id="cb23-45"><a href="#cb23-45" tabindex="-1"></a>                            <span class="st">"gene_start"</span><span class="sc">:</span> start,</span>
<span id="cb23-46"><a href="#cb23-46" tabindex="-1"></a>                            <span class="st">"gene_end"</span><span class="sc">:</span> end,</span>
<span id="cb23-47"><a href="#cb23-47" tabindex="-1"></a>                            <span class="st">"chr"</span><span class="sc">:</span> chrom,</span>
<span id="cb23-48"><a href="#cb23-48" tabindex="-1"></a>                            <span class="st">"biotype"</span><span class="sc">:</span> gene_type</span>
<span id="cb23-49"><a href="#cb23-49" tabindex="-1"></a>                        }</span>
<span id="cb23-50"><a href="#cb23-50" tabindex="-1"></a></span>
<span id="cb23-51"><a href="#cb23-51" tabindex="-1"></a>except FileNotFoundError<span class="sc">:</span></span>
<span id="cb23-52"><a href="#cb23-52" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"File not found: {}"</span><span class="fu">.format</span>(file_name))</span>
<span id="cb23-53"><a href="#cb23-53" tabindex="-1"></a>    <span class="fu">sys.exit</span>(<span class="dv">1</span>)</span>
<span id="cb23-54"><a href="#cb23-54" tabindex="-1"></a></span>
<span id="cb23-55"><a href="#cb23-55" tabindex="-1"></a><span class="co"># Print the output                                                                                                                                                                       </span></span>
<span id="cb23-56"><a href="#cb23-56" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"</span><span class="sc">\t</span><span class="st">"</span><span class="fu">.join</span>([<span class="st">"chrom"</span>, <span class="st">"gene_start"</span>, <span class="st">"gene_end"</span>, <span class="st">"gene_symbol"</span>, <span class="st">"biotype"</span>, <span class="st">"exon_chromstart"</span>, <span class="st">"exon_chromend"</span>]))</span>
<span id="cb23-57"><a href="#cb23-57" tabindex="-1"></a><span class="cf">for</span> gene, data <span class="cf">in</span> <span class="fu">genes.items</span>()<span class="sc">:</span></span>
<span id="cb23-58"><a href="#cb23-58" tabindex="-1"></a>    <span class="cf">if</span> not gene<span class="sc">:</span></span>
<span id="cb23-59"><a href="#cb23-59" tabindex="-1"></a>        continue</span>
<span id="cb23-60"><a href="#cb23-60" tabindex="-1"></a>    exon_starts <span class="ot">=</span> <span class="st">","</span><span class="fu">.join</span>(<span class="fu">data.get</span>(<span class="st">"exon_chromstart"</span>, []))</span>
<span id="cb23-61"><a href="#cb23-61" tabindex="-1"></a>    exon_ends <span class="ot">=</span> <span class="st">","</span><span class="fu">.join</span>(<span class="fu">data.get</span>(<span class="st">"exon_chromend"</span>, []))</span>
<span id="cb23-62"><a href="#cb23-62" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"</span><span class="sc">\t</span><span class="st">"</span><span class="fu">.join</span>([data[<span class="st">"chr"</span>], data[<span class="st">"gene_start"</span>], data[<span class="st">"gene_end"</span>], gene, data[<span class="st">"biotype"</span>], exon_starts, exon_ends]))</span></code></pre></div>
</details><p><br></p>
<p>Then from within your R editor do:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mm39</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="st">"Mus_musculus.GRCm39.111.tsv"</span>, sep<span class="op">=</span><span class="st">"\t"</span>,header<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p>Then optionally to save the data in a more compact .rda format:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">mm39</span>, file<span class="op">=</span><span class="st">"mus_musculus.rda"</span>, compress<span class="op">=</span><span class="st">'xz'</span><span class="op">)</span></span>
<span><span class="co">#and then to load from the .rda file, do:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"mus_musculus.rda"</span><span class="op">)</span></span></code></pre></div>
<p>Then to use the mouse build with <em>topr</em>, assign <code>mm39</code> to the build argument in <em>topr’s</em> functions.</p>
<p>Examples (note that CD_UKBB are human association results, and only used here as a proof of concept):</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, annotate<span class="op">=</span><span class="fl">1e-14</span>, build<span class="op">=</span><span class="va">mm39</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/regionplot.html">regionplot</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, gene<span class="op">=</span><span class="st">"Cps1"</span>, build<span class="op">=</span><span class="va">mm39</span><span class="op">)</span></span>
<span><span class="va">CD_UKBB</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="reference/get_lead_snps.html">get_lead_snps</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="reference/annotate_with_nearest_gene.html">annotate_with_nearest_gene</a></span><span class="op">(</span>build<span class="op">=</span><span class="va">mm39</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/get_gene_coords.html">get_gene_coords</a></span><span class="op">(</span><span class="st">"Cps1"</span>, build<span class="op">=</span><span class="va">mm39</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
</div>
<div class="section level4">
<h4 id="how-to-color-specific-peaks-on-the-manhattan-plot">How to color specific peaks on the Manhattan plot<a class="anchor" aria-label="anchor" href="#how-to-color-specific-peaks-on-the-manhattan-plot"></a>
</h4>
<hr>
<p>I use the inbuilt <code>CD_UKBB</code> dataset in this example and create two dataframes containing hypothetically known and novel loci.</p>
<p>I start by annotating the variants in the <code>CD_UKBB</code> dataset so they can be extracted based on their nearest gene annotation.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">CD_UKBB_annotated</span> <span class="op">&lt;-</span> <span class="va">CD_UKBB</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="reference/annotate_with_nearest_gene.html">annotate_with_nearest_gene</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Next I create two dataframes, one including an example of hypothetically <code>known</code> variants and the other an example of hypothetically <code>novel</code> variants.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">known</span> <span class="op">&lt;-</span> <span class="va">CD_UKBB_annotated</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Gene_Symbol</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"C1orf141"</span>,<span class="st">"IL23R"</span>,<span class="st">"NOD2"</span>,<span class="st">"CYLD-AS1"</span>,<span class="st">"IKZF1"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">novel</span> <span class="op">&lt;-</span> <span class="va">CD_UKBB_annotated</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Gene_Symbol</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"JAK2"</span>,<span class="st">"TTC33"</span>,<span class="st">"ATG16L1"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Now I can call the <code>manhattan</code> function with a list containing the 3 datasets (<code>CD_UKBB, known, novel</code>) and assign a color to each dataset (darkgrey, blue, red), where novel loci are displayed in red.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">known</span>, <span class="va">novel</span><span class="op">)</span>, color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"darkgrey"</span>,<span class="st">"blue"</span>,<span class="st">"red"</span><span class="op">)</span>, annotate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1e-100</span>,<span class="fl">5e-08</span>,<span class="fl">5e-08</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/manhattan_colored_peaks.png" alt="topr GIF" width="100%"></p>
<p>Note that here I only want to annotate the top variants in the <code>novel</code> and <code>known</code> datasets, and therefore I set the first p-value (for the first dataset CD_UKBB) assigned to the <code>annotate</code> argument to a very low number (1e-100) (so that nothing gets annotated in the CD_UKBB dataset).</p>
<p>As can be seen on the plot, known and novel peaks on even numbered chromosomes are displayed in a lighter color of blue and red. This can be altered with the <code>even_no_chr_lightness</code> argument and shown next.</p>
<p><br></p>
<div class="section level5">
<h5 id="use-the-same-color-of-red-and-blue-for-all-the-peaks">Use the same color of red and blue for all the peaks<a class="anchor" aria-label="anchor" href="#use-the-same-color-of-red-and-blue-for-all-the-peaks"></a>
</h5>
<p>By default, association points on even numbered chromosomes (2,4,6 etc) are displayed in a slightly lighter color compared to the association points displayed on odd numbered chromosomes (1,3,5 etc).</p>
<p>This can be controlled with the <code>even_no_chr_lightness</code> argument which is set to 0.8 by default. If this argument is set to 0.5 for a given dataset and color, the same color will be displayed on all chromosomes for this dataset. A value below 0.5 will result in a darker color displayed for even numbered chromosomes, whereas a value above 0.5 results in a lighter color.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">known</span>, <span class="va">novel</span><span class="op">)</span>, color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"darkgrey"</span>,<span class="st">"blue"</span>,<span class="st">"red"</span><span class="op">)</span>, annotate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1e-100</span>,<span class="fl">5e-08</span>,<span class="fl">5e-08</span> <span class="op">)</span>, even_no_chr_lightness <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.8</span>,<span class="fl">0.5</span>,<span class="fl">0.5</span><span class="op">)</span>, legend_labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD_UKBB"</span>,<span class="st">"Known loci"</span>,<span class="st">"Novel loci"</span><span class="op">)</span>, label_color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/manhattan_colored_peaks2.png" alt="topr GIF" width="100%"></p>
<p><br></p>
</div>
</div>
<div class="section level4">
<h4 id="manhattanextra">manhattanExtra<a class="anchor" aria-label="anchor" href="#manhattanextra"></a>
</h4>
<hr>
<p><em>Contribution from Kyle Scott <a href="https://github.com/kscott-1" class="external-link">kscott-1</a></em></p>
<p>Create a Manhattan plot highlighting genome-wide significant and suggestive loci using the <code>manhattanExtra</code> function.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/manhattanExtra.html">manhattanExtra</a></span><span class="op">(</span><span class="va">UC_UKBB</span>, flank_size <span class="op">=</span> <span class="fl">1e6</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/manhattanExtra.png" alt="manhattanExtra" width="100%"></p>
<p><br></p>
</div>
<div class="section level4">
<h4 id="locuszoom-example">Locuszoom example<a class="anchor" aria-label="anchor" href="#locuszoom-example"></a>
</h4>
<hr>
<p><em>Contribution from Kyle Scott <a href="https://github.com/kscott-1" class="external-link">kscott-1</a></em></p>
<p>Extract the snp of interest <em>rs7713270</em>, from the inbuilt CD_UKBB dataset.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lead_snp</span> <span class="op">&lt;-</span> <span class="va">CD_UKBB</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ID</span> <span class="op">==</span> <span class="st">"rs7713270"</span><span class="op">)</span> </span>
<span><span class="va">snp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">lead_snp</span><span class="op">$</span><span class="va">CHROM</span>,<span class="st">":"</span>,<span class="va">lead_snp</span><span class="op">$</span><span class="va">POS</span><span class="op">)</span></span></code></pre></div>
<div class="section level5">
<h5 id="retrieve-the-r2-values-with-ldroxy">Retrieve the R2 values with LDroxy<a class="anchor" aria-label="anchor" href="#retrieve-the-r2-values-with-ldroxy"></a>
</h5>
<p>Run <code>LDproxy</code> to get variants correlated with <em>rs7713270</em>. Use the European (EUR) 1000 Genomes Project population and genome build GRCh38.</p>
<p>Note! To be able to run <code>LDproxy</code>, you have to register and get a token (see <a href="https://ldlink.nih.gov/?tab=apiaccess" class="external-link uri">https://ldlink.nih.gov/?tab=apiaccess</a>).</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">SNP</span> <span class="op">&lt;-</span> <span class="fu">LDproxy</span><span class="op">(</span></span>
<span>  snp <span class="op">=</span> <span class="va">snp</span>, pop <span class="op">=</span> <span class="st">"EUR"</span>, r2d <span class="op">=</span> <span class="st">"r2"</span>,</span>
<span>  token <span class="op">=</span> <span class="st">"NULL"</span>, genome_build <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"grch"</span>, <span class="st">"38"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">LD.link</span> <span class="op">&lt;-</span> <span class="va">SNP</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>CHROM <span class="op">=</span> <span class="fu">str_extract</span><span class="op">(</span><span class="va">Coord</span>, <span class="st">"chr[^:]*"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>POS <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu">str_extract</span><span class="op">(</span><span class="va">Coord</span>, <span class="st">"(?&lt;=:)[0-9]*"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>  </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">CHROM</span>, <span class="va">POS</span>, <span class="va">R2</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="plot-with-the-locuszoom-function">Plot with the locuszoom function<a class="anchor" aria-label="anchor" href="#plot-with-the-locuszoom-function"></a>
</h5>
<p>Join the output from LDproxy with the CD_UKBB dataset to get the p-values for plotting.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">snps.ld</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">LD.link</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CHROM"</span>, <span class="st">"POS"</span><span class="op">)</span><span class="op">)</span> </span>
<span><span class="fu"><a href="reference/locuszoom.html">locuszoom</a></span><span class="op">(</span><span class="va">snps.ld</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/locuszoom.png" alt="topr GIF" width="80%"></p>
</div>
</div>
<div class="section level4">
<h4 id="get-help">Get help:<a class="anchor" aria-label="anchor" href="#get-help"></a>
</h4>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">?</span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">?</span><span class="fu"><a href="reference/regionplot.html">regionplot</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">?</span><span class="fu"><a href="reference/locuszoom.html">locuszoom</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=topr" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/totajuliusd/topr/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/totajuliusd/topr/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>LGPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing topr</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Thorhildur Juliusdottir <br><small class="roles"> Copyright holder, author, maintainer </small>  </li>
<li>Andri Stefansson <br><small class="roles"> Author </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Thorhildur Juliusdottir, Andri Stefansson.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
