<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Create a plot comparing variant effects in two datasets — effectplot • topr</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Create a plot comparing variant effects in two datasets — effectplot"><meta name="description" content="effectplot()"><meta property="og:description" content="effectplot()"><meta property="og:image" content="https://kscott-1.github.io/topr/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">topr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/topr.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/locuszoom.html">Locuszoom plot examples</a></li>
    <li><a class="dropdown-item" href="../articles/manhattanExtra.html">Extending manhattan plots with manhattanExtra</a></li>
    <li><a class="dropdown-item" href="../articles/moreManhattan.html">More manhattan examples</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/totajuliusd/topr/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Create a plot comparing variant effects in two datasets</h1>
      <small class="dont-index">Source: <a href="https://github.com/totajuliusd/topr/blob/doc_testing/R/effectplot.R" class="external-link"><code>R/effectplot.R</code></a></small>
      <div class="d-none name"><code>effectplot.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>effectplot()</code></p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">effectplot</span><span class="op">(</span></span>
<span>  <span class="va">df</span>,</span>
<span>  pheno_x <span class="op">=</span> <span class="st">"x_pheno"</span>,</span>
<span>  pheno_y <span class="op">=</span> <span class="st">"y_pheno"</span>,</span>
<span>  annotate_with <span class="op">=</span> <span class="st">"Gene_Symbol"</span>,</span>
<span>  thresh <span class="op">=</span> <span class="fl">5e-08</span>,</span>
<span>  ci_thresh <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  gene_label_thresh <span class="op">=</span> <span class="fl">5e-08</span>,</span>
<span>  color <span class="op">=</span> <span class="fu"><a href="get_topr_colors.html">get_topr_colors</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>  scale <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  build <span class="op">=</span> <span class="fl">38</span>,</span>
<span>  label_fontface <span class="op">=</span> <span class="st">"italic"</span>,</span>
<span>  label_family <span class="op">=</span> <span class="st">""</span>,</span>
<span>  nudge_y <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>  nudge_x <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>  size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  segment.size <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  segment.linetype <span class="op">=</span> <span class="st">"solid"</span>,</span>
<span>  segment.color <span class="op">=</span> <span class="st">"transparent"</span>,</span>
<span>  angle <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  axis_text_size <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  axis_title_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  title_text_size <span class="op">=</span> <span class="fl">13</span>,</span>
<span>  subtitle_text_size <span class="op">=</span> <span class="fl">11</span>,</span>
<span>  gene_label_size <span class="op">=</span> <span class="fl">3.2</span>,</span>
<span>  snpset_thresh <span class="op">=</span> <span class="fl">5e-08</span>,</span>
<span>  snpset_region_size <span class="op">=</span> <span class="fl">1e+06</span>,</span>
<span>  max.overlaps <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  annotate <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  label_color <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-df">df<a class="anchor" aria-label="anchor" href="#arg-df"></a></dt>
<dd><p>The input dataframe (snpset) containing one row per variant and P values (P1 and P2) and effects (E1 and E2) from two datasets/phenotypes OR a list containing two datasets.</p></dd>


<dt id="arg-pheno-x">pheno_x<a class="anchor" aria-label="anchor" href="#arg-pheno-x"></a></dt>
<dd><p>A string representing the name of the phenotype whose effect is plotted on the x axis</p></dd>


<dt id="arg-pheno-y">pheno_y<a class="anchor" aria-label="anchor" href="#arg-pheno-y"></a></dt>
<dd><p>A string representing the name of the phenotype whose effect is plotted on the y axis</p></dd>


<dt id="arg-annotate-with">annotate_with<a class="anchor" aria-label="anchor" href="#arg-annotate-with"></a></dt>
<dd><p>A string, The name of the column that contains the label for the datapoints (default value is Gene_Symbol)</p></dd>


<dt id="arg-thresh">thresh<a class="anchor" aria-label="anchor" href="#arg-thresh"></a></dt>
<dd><p>A number. Threshold cutoff, datapoints with P2 below this threshold are shown as filled circles whereas datapoints with P2 above this threshold are shown as open circles</p></dd>


<dt id="arg-ci-thresh">ci_thresh<a class="anchor" aria-label="anchor" href="#arg-ci-thresh"></a></dt>
<dd><p>A number.Show the confidence intervals if the P-value is below this threshold</p></dd>


<dt id="arg-gene-label-thresh">gene_label_thresh<a class="anchor" aria-label="anchor" href="#arg-gene-label-thresh"></a></dt>
<dd><p>Deprecated: A number, label datapoints with P2 below this threshold</p></dd>


<dt id="arg-color">color<a class="anchor" aria-label="anchor" href="#arg-color"></a></dt>
<dd><p>A string, default value is the first of the topr colors</p></dd>


<dt id="arg-scale">scale<a class="anchor" aria-label="anchor" href="#arg-scale"></a></dt>
<dd><p>A number, to change the size of the title and axes labels and ticks at the same time (default : 1)</p></dd>


<dt id="arg-build">build<a class="anchor" aria-label="anchor" href="#arg-build"></a></dt>
<dd><p>A number representing the genome build or a data frame. Set to 37 to change to build (GRCh37). The default is build 38 (GRCh38).</p></dd>


<dt id="arg-label-fontface">label_fontface<a class="anchor" aria-label="anchor" href="#arg-label-fontface"></a></dt>
<dd><p>A string or a vector of strings. Label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p></dd>


<dt id="arg-label-family">label_family<a class="anchor" aria-label="anchor" href="#arg-label-family"></a></dt>
<dd><p>A string or a vector of strings. Label font name (default ggrepel argument is "")</p></dd>


<dt id="arg-nudge-y">nudge_y<a class="anchor" aria-label="anchor" href="#arg-nudge-y"></a></dt>
<dd><p>A number to horizontally adjust the starting position of each gene label (this is a ggrepel parameter)</p></dd>


<dt id="arg-nudge-x">nudge_x<a class="anchor" aria-label="anchor" href="#arg-nudge-x"></a></dt>
<dd><p>A number to vertically adjust the starting position of each gene label (this is a ggrepel parameter)</p></dd>


<dt id="arg-size">size<a class="anchor" aria-label="anchor" href="#arg-size"></a></dt>
<dd><p>A number or a vector of numbers, setting the size of the plot points (default: <code>size=1.2</code>)</p></dd>


<dt id="arg-segment-size">segment.size<a class="anchor" aria-label="anchor" href="#arg-segment-size"></a></dt>
<dd><p>line segment color (ggrepel argument)</p></dd>


<dt id="arg-segment-linetype">segment.linetype<a class="anchor" aria-label="anchor" href="#arg-segment-linetype"></a></dt>
<dd><p>line segment solid, dashed, etc.(ggrepel argument)</p></dd>


<dt id="arg-segment-color">segment.color<a class="anchor" aria-label="anchor" href="#arg-segment-color"></a></dt>
<dd><p>line segment thickness (ggrepel argument)</p></dd>


<dt id="arg-angle">angle<a class="anchor" aria-label="anchor" href="#arg-angle"></a></dt>
<dd><p>A number, the angle of the text label</p></dd>


<dt id="arg-title">title<a class="anchor" aria-label="anchor" href="#arg-title"></a></dt>
<dd><p>A string to set the plot title</p></dd>


<dt id="arg-axis-text-size">axis_text_size<a class="anchor" aria-label="anchor" href="#arg-axis-text-size"></a></dt>
<dd><p>A number, size of the x and y axes tick labels (default: 12)</p></dd>


<dt id="arg-axis-title-size">axis_title_size<a class="anchor" aria-label="anchor" href="#arg-axis-title-size"></a></dt>
<dd><p>A number, size of the x and y title labels (default: 12)</p></dd>


<dt id="arg-title-text-size">title_text_size<a class="anchor" aria-label="anchor" href="#arg-title-text-size"></a></dt>
<dd><p>A number, size of the plot title (default: 13)</p></dd>


<dt id="arg-subtitle-text-size">subtitle_text_size<a class="anchor" aria-label="anchor" href="#arg-subtitle-text-size"></a></dt>
<dd><p>A number setting the text size of the subtitle (default: 11)</p></dd>


<dt id="arg-gene-label-size">gene_label_size<a class="anchor" aria-label="anchor" href="#arg-gene-label-size"></a></dt>
<dd><p>A number setting the size of the gene labels shown at the bottom of the plot</p></dd>


<dt id="arg-snpset-thresh">snpset_thresh<a class="anchor" aria-label="anchor" href="#arg-snpset-thresh"></a></dt>
<dd><p>A number representing the threshold used to create the snpset used for plotting (Only applicable if the input dataframe is a list containing two datasets)</p></dd>


<dt id="arg-snpset-region-size">snpset_region_size<a class="anchor" aria-label="anchor" href="#arg-snpset-region-size"></a></dt>
<dd><p>A number representing the region size to use when creating the snpset used for plotting (Only applicable if the input dataframe is a list containing two datasets)</p></dd>


<dt id="arg-max-overlaps">max.overlaps<a class="anchor" aria-label="anchor" href="#arg-max-overlaps"></a></dt>
<dd><p>Exclude text labels that overlap too many things. Defaults to 10 (ggrepel argument)</p></dd>


<dt id="arg-annotate">annotate<a class="anchor" aria-label="anchor" href="#arg-annotate"></a></dt>
<dd><p>A number, label datapoints with p-value below below this number (in the second df) by their nearest gene</p></dd>


<dt id="arg-label-color">label_color<a class="anchor" aria-label="anchor" href="#arg-label-color"></a></dt>
<dd><p>A string or a vector of strings. To change the color of the gene or variant labels</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>ggplot object</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="fu">effectplot</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Thorhildur Juliusdottir, Andri Stefansson.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

