<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Create a Manhattan plot — manhattan • topr</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Create a Manhattan plot — manhattan"><meta name="description" content="manhattan() displays association results for the entire genome on a Manhattan plot.
Required parameter is at least one dataset (dataframe) containing the association data (with columns CHROM,POS,P in upper or lowercase)
All other input parameters are optional"><meta property="og:description" content="manhattan() displays association results for the entire genome on a Manhattan plot.
Required parameter is at least one dataset (dataframe) containing the association data (with columns CHROM,POS,P in upper or lowercase)
All other input parameters are optional"><meta property="og:image" content="https://kscott-1.github.io/topr/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">topr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/topr.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/locuszoom.html">Locuszoom plot examples</a></li>
    <li><a class="dropdown-item" href="../articles/manhattanExtra.html">Extending manhattan plots with manhattanExtra</a></li>
    <li><a class="dropdown-item" href="../articles/moreManhattan.html">More manhattan examples</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/totajuliusd/topr/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Create a Manhattan plot</h1>
      <small class="dont-index">Source: <a href="https://github.com/totajuliusd/topr/blob/doc_testing/R/manhattan.R" class="external-link"><code>R/manhattan.R</code></a></small>
      <div class="d-none name"><code>manhattan.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>manhattan()</code> displays association results for the entire genome on a Manhattan plot.
Required parameter is at least one dataset (dataframe) containing the association data (with columns <code>CHROM,POS,P</code> in upper or lowercase)</p>
<p>All other input parameters are optional</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">manhattan</span><span class="op">(</span></span>
<span>  <span class="va">df</span>,</span>
<span>  ntop <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  title <span class="op">=</span> <span class="st">""</span>,</span>
<span>  annotate <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  color <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sign_thresh <span class="op">=</span> <span class="fl">5e-08</span>,</span>
<span>  sign_thresh_color <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>  sign_thresh_label_size <span class="op">=</span> <span class="fl">3.5</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">3.5</span>,</span>
<span>  size <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  shape <span class="op">=</span> <span class="fl">19</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  highlight_genes_color <span class="op">=</span> <span class="st">"darkred"</span>,</span>
<span>  highlight_genes_ypos <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  axis_text_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  axis_title_size <span class="op">=</span> <span class="fl">14</span>,</span>
<span>  title_text_size <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  legend_title_size <span class="op">=</span> <span class="fl">13</span>,</span>
<span>  legend_text_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  protein_coding_only <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  angle <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  legend_labels <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  chr <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  annotate_with <span class="op">=</span> <span class="st">"Gene_Symbol"</span>,</span>
<span>  region_size <span class="op">=</span> <span class="fl">2e+07</span>,</span>
<span>  legend_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  legend_position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>  nudge_x <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  nudge_y <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>  xmin <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  xmax <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ymin <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ymax <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  highlight_genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  label_color <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  legend_nrow <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_label_size <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_label_angle <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  scale <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  show_legend <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sign_thresh_linetype <span class="op">=</span> <span class="st">"dashed"</span>,</span>
<span>  sign_thresh_size <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  rsids <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  rsids_color <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  rsids_with_vline <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  annotate_with_vline <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  shades_color <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  shades_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  segment.size <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  segment.color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>  segment.linetype <span class="op">=</span> <span class="st">"dashed"</span>,</span>
<span>  max.overlaps <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  label_fontface <span class="op">=</span> <span class="st">"plain"</span>,</span>
<span>  label_family <span class="op">=</span> <span class="st">""</span>,</span>
<span>  gene_label_fontface <span class="op">=</span> <span class="st">"plain"</span>,</span>
<span>  gene_label_family <span class="op">=</span> <span class="st">""</span>,</span>
<span>  build <span class="op">=</span> <span class="fl">38</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  label_alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  shades_line_alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  vline <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  vline_color <span class="op">=</span> <span class="st">"grey"</span>,</span>
<span>  vline_linetype <span class="op">=</span> <span class="st">"dashed"</span>,</span>
<span>  vline_alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  vline_size <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  region <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  theme_grey <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  xaxis_label <span class="op">=</span> <span class="st">"Chromosome"</span>,</span>
<span>  use_shades <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  even_no_chr_lightness <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  get_chr_lengths_from_data <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  log_trans_p <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  chr_ticknames <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  show_all_chrticks <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  hide_chrticks_from_pos <span class="op">=</span> <span class="fl">17</span>,</span>
<span>  hide_chrticks_to_pos <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  hide_every_nth_chrtick <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  downsample_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  downsample_prop <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-df">df<a class="anchor" aria-label="anchor" href="#arg-df"></a></dt>
<dd><p>Dataframe or a list of dataframes (required columns are <code>CHROM,POS,P</code>), in upper- or lowercase) of association results.</p></dd>


<dt id="arg-ntop">ntop<a class="anchor" aria-label="anchor" href="#arg-ntop"></a></dt>
<dd><p>An integer, number of datasets (GWAS results) to show on the top plot</p></dd>


<dt id="arg-title">title<a class="anchor" aria-label="anchor" href="#arg-title"></a></dt>
<dd><p>A string to set the plot title</p></dd>


<dt id="arg-annotate">annotate<a class="anchor" aria-label="anchor" href="#arg-annotate"></a></dt>
<dd><p>A number (p-value). Display annotation for variants with p-values below this threshold</p></dd>


<dt id="arg-color">color<a class="anchor" aria-label="anchor" href="#arg-color"></a></dt>
<dd><p>A string or a vector of strings, for setting the color of the datapoints on the plot</p></dd>


<dt id="arg-sign-thresh">sign_thresh<a class="anchor" aria-label="anchor" href="#arg-sign-thresh"></a></dt>
<dd><p>A number or vector of numbers, setting the horizontal significance threshold (default: <code>sign_thresh=5e-8</code>). Set to NULL to hide the significance threshold.</p></dd>


<dt id="arg-sign-thresh-color">sign_thresh_color<a class="anchor" aria-label="anchor" href="#arg-sign-thresh-color"></a></dt>
<dd><p>A string or vector of strings to set the color/s of the significance threshold/s</p></dd>


<dt id="arg-sign-thresh-label-size">sign_thresh_label_size<a class="anchor" aria-label="anchor" href="#arg-sign-thresh-label-size"></a></dt>
<dd><p>A number setting the text size of the label for the significance thresholds (default text size is 3.5)</p></dd>


<dt id="arg-label-size">label_size<a class="anchor" aria-label="anchor" href="#arg-label-size"></a></dt>
<dd><p>An number to set the size of the plot labels (default: <code>label_size=3</code>)</p></dd>


<dt id="arg-size">size<a class="anchor" aria-label="anchor" href="#arg-size"></a></dt>
<dd><p>A number or a vector of numbers, setting the size of the plot points (default: <code>size=1.2</code>)</p></dd>


<dt id="arg-shape">shape<a class="anchor" aria-label="anchor" href="#arg-shape"></a></dt>
<dd><p>A number of a vector of numbers setting the shape of the plotted points</p></dd>


<dt id="arg-alpha">alpha<a class="anchor" aria-label="anchor" href="#arg-alpha"></a></dt>
<dd><p>A number or a vector of numbers setting the transparency of the plotted points</p></dd>


<dt id="arg-highlight-genes-color">highlight_genes_color<a class="anchor" aria-label="anchor" href="#arg-highlight-genes-color"></a></dt>
<dd><p>A string, color for the highlighted genes (default: darkred)</p></dd>


<dt id="arg-highlight-genes-ypos">highlight_genes_ypos<a class="anchor" aria-label="anchor" href="#arg-highlight-genes-ypos"></a></dt>
<dd><p>An integer, controlling where on the y-axis the highlighted genes are placed (default value is 1)</p></dd>


<dt id="arg-axis-text-size">axis_text_size<a class="anchor" aria-label="anchor" href="#arg-axis-text-size"></a></dt>
<dd><p>A number, size of the x and y axes tick labels (default: 12)</p></dd>


<dt id="arg-axis-title-size">axis_title_size<a class="anchor" aria-label="anchor" href="#arg-axis-title-size"></a></dt>
<dd><p>A number, size of the x and y title labels (default: 12)</p></dd>


<dt id="arg-title-text-size">title_text_size<a class="anchor" aria-label="anchor" href="#arg-title-text-size"></a></dt>
<dd><p>A number, size of the plot title (default: 13)</p></dd>


<dt id="arg-legend-title-size">legend_title_size<a class="anchor" aria-label="anchor" href="#arg-legend-title-size"></a></dt>
<dd><p>A number, size of the legend title</p></dd>


<dt id="arg-legend-text-size">legend_text_size<a class="anchor" aria-label="anchor" href="#arg-legend-text-size"></a></dt>
<dd><p>A number, size of the legend text</p></dd>


<dt id="arg-protein-coding-only">protein_coding_only<a class="anchor" aria-label="anchor" href="#arg-protein-coding-only"></a></dt>
<dd><p>A logical scalar, if TRUE, only protein coding genes are used for annotation</p></dd>


<dt id="arg-angle">angle<a class="anchor" aria-label="anchor" href="#arg-angle"></a></dt>
<dd><p>A number, the angle of the text label</p></dd>


<dt id="arg-legend-labels">legend_labels<a class="anchor" aria-label="anchor" href="#arg-legend-labels"></a></dt>
<dd><p>A string or vector of strings representing legend labels for the input datasets</p></dd>


<dt id="arg-chr">chr<a class="anchor" aria-label="anchor" href="#arg-chr"></a></dt>
<dd><p>A string or integer, the chromosome to plot (i.e. chr15), only required if the input dataframe contains results from more than one chromosome</p></dd>


<dt id="arg-annotate-with">annotate_with<a class="anchor" aria-label="anchor" href="#arg-annotate-with"></a></dt>
<dd><p>A string. Annotate the variants with either Gene_Symbol or ID (default: "Gene_Symbol")</p></dd>


<dt id="arg-region-size">region_size<a class="anchor" aria-label="anchor" href="#arg-region-size"></a></dt>
<dd><p>An integer (default = 20000000) (or a string represented as 200kb or 2MB) indicating the window size for variant labeling. Increase this number for sparser annotation and decrease for denser annotation.</p></dd>


<dt id="arg-legend-name">legend_name<a class="anchor" aria-label="anchor" href="#arg-legend-name"></a></dt>
<dd><p>A string, use to change the name of the legend (default: None)</p></dd>


<dt id="arg-legend-position">legend_position<a class="anchor" aria-label="anchor" href="#arg-legend-position"></a></dt>
<dd><p>A string, top,bottom,left or right</p></dd>


<dt id="arg-nudge-x">nudge_x<a class="anchor" aria-label="anchor" href="#arg-nudge-x"></a></dt>
<dd><p>A number to vertically adjust the starting position of each gene label (this is a ggrepel parameter)</p></dd>


<dt id="arg-nudge-y">nudge_y<a class="anchor" aria-label="anchor" href="#arg-nudge-y"></a></dt>
<dd><p>A number to horizontally adjust the starting position of each gene label (this is a ggrepel parameter)</p></dd>


<dt id="arg-xmin-xmax">xmin, xmax<a class="anchor" aria-label="anchor" href="#arg-xmin-xmax"></a></dt>
<dd><p>Integer, setting the chromosomal range to display on the x-axis</p></dd>


<dt id="arg-ymin-ymax">ymin, ymax<a class="anchor" aria-label="anchor" href="#arg-ymin-ymax"></a></dt>
<dd><p>Integer, min and max of the y-axis, (default values: <code>ymin=0, ymax=max(-log10(df$P))</code>)</p></dd>


<dt id="arg-highlight-genes">highlight_genes<a class="anchor" aria-label="anchor" href="#arg-highlight-genes"></a></dt>
<dd><p>A string or vector of strings, gene or genes to highlight at the bottom of the plot</p></dd>


<dt id="arg-label-color">label_color<a class="anchor" aria-label="anchor" href="#arg-label-color"></a></dt>
<dd><p>A string or a vector of strings. To change the color of the gene or variant labels</p></dd>


<dt id="arg-legend-nrow">legend_nrow<a class="anchor" aria-label="anchor" href="#arg-legend-nrow"></a></dt>
<dd><p>An integer, sets the number of rows allowed for the legend labels</p></dd>


<dt id="arg-gene-label-size">gene_label_size<a class="anchor" aria-label="anchor" href="#arg-gene-label-size"></a></dt>
<dd><p>A number setting the size of the gene labels shown at the bottom of the plot</p></dd>


<dt id="arg-gene-label-angle">gene_label_angle<a class="anchor" aria-label="anchor" href="#arg-gene-label-angle"></a></dt>
<dd><p>A number setting the angle of the gene label shown at the bottom of the plot (default: 0)</p></dd>


<dt id="arg-scale">scale<a class="anchor" aria-label="anchor" href="#arg-scale"></a></dt>
<dd><p>A number, to change the size of the title and axes labels and ticks at the same time (default : 1)</p></dd>


<dt id="arg-show-legend">show_legend<a class="anchor" aria-label="anchor" href="#arg-show-legend"></a></dt>
<dd><p>A logical scalar, set to FALSE to hide the legend (default : TRUE)</p></dd>


<dt id="arg-sign-thresh-linetype">sign_thresh_linetype<a class="anchor" aria-label="anchor" href="#arg-sign-thresh-linetype"></a></dt>
<dd><p>A string, the line-type of the horizontal significance threshold (default : dashed)</p></dd>


<dt id="arg-sign-thresh-size">sign_thresh_size<a class="anchor" aria-label="anchor" href="#arg-sign-thresh-size"></a></dt>
<dd><p>A number, sets the size of the horizontal significance threshold line (default : 1)</p></dd>


<dt id="arg-rsids">rsids<a class="anchor" aria-label="anchor" href="#arg-rsids"></a></dt>
<dd><p>A string (rsid) or vector of strings to highlight on the plot, e.g. <code>rsids=c("rs1234, rs45898")</code></p></dd>


<dt id="arg-rsids-color">rsids_color<a class="anchor" aria-label="anchor" href="#arg-rsids-color"></a></dt>
<dd><p>A string, the color of the variants in variants_id (default color is red)</p></dd>


<dt id="arg-rsids-with-vline">rsids_with_vline<a class="anchor" aria-label="anchor" href="#arg-rsids-with-vline"></a></dt>
<dd><p>A string (rsid) or vector of strings to highlight on the plot with their rsids and vertical lines further highlighting their positions</p></dd>


<dt id="arg-annotate-with-vline">annotate_with_vline<a class="anchor" aria-label="anchor" href="#arg-annotate-with-vline"></a></dt>
<dd><p>A number (p-value). Display annotation and vertical lines for variants with p-values below this threshold</p></dd>


<dt id="arg-shades-color">shades_color<a class="anchor" aria-label="anchor" href="#arg-shades-color"></a></dt>
<dd><p>The color of the rectangles (shades) representing the different chromosomes on the Manhattan plot</p></dd>


<dt id="arg-shades-alpha">shades_alpha<a class="anchor" aria-label="anchor" href="#arg-shades-alpha"></a></dt>
<dd><p>The transparency (alpha) of the rectangles (shades)</p></dd>


<dt id="arg-segment-size">segment.size<a class="anchor" aria-label="anchor" href="#arg-segment-size"></a></dt>
<dd><p>line segment color (ggrepel argument)</p></dd>


<dt id="arg-segment-color">segment.color<a class="anchor" aria-label="anchor" href="#arg-segment-color"></a></dt>
<dd><p>line segment thickness (ggrepel argument)</p></dd>


<dt id="arg-segment-linetype">segment.linetype<a class="anchor" aria-label="anchor" href="#arg-segment-linetype"></a></dt>
<dd><p>line segment solid, dashed, etc.(ggrepel argument)</p></dd>


<dt id="arg-max-overlaps">max.overlaps<a class="anchor" aria-label="anchor" href="#arg-max-overlaps"></a></dt>
<dd><p>Exclude text labels that overlap too many things. Defaults to 10 (ggrepel argument)</p></dd>


<dt id="arg-label-fontface">label_fontface<a class="anchor" aria-label="anchor" href="#arg-label-fontface"></a></dt>
<dd><p>A string or a vector of strings. Label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p></dd>


<dt id="arg-label-family">label_family<a class="anchor" aria-label="anchor" href="#arg-label-family"></a></dt>
<dd><p>A string or a vector of strings. Label font name (default ggrepel argument is "")</p></dd>


<dt id="arg-gene-label-fontface">gene_label_fontface<a class="anchor" aria-label="anchor" href="#arg-gene-label-fontface"></a></dt>
<dd><p>Gene label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p></dd>


<dt id="arg-gene-label-family">gene_label_family<a class="anchor" aria-label="anchor" href="#arg-gene-label-family"></a></dt>
<dd><p>Gene label font name (default ggrepel argument is "")</p></dd>


<dt id="arg-build">build<a class="anchor" aria-label="anchor" href="#arg-build"></a></dt>
<dd><p>A number representing the genome build or a data frame. Set to 37 to change to build (GRCh37). The default is build 38 (GRCh38).</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>A logical scalar (default: NULL). Set to FALSE to suppress printed messages</p></dd>


<dt id="arg-label-alpha">label_alpha<a class="anchor" aria-label="anchor" href="#arg-label-alpha"></a></dt>
<dd><p>An number or vector of numbers to set the transparency of the plot labels (default: <code>label_alpha=1</code>)</p></dd>


<dt id="arg-shades-line-alpha">shades_line_alpha<a class="anchor" aria-label="anchor" href="#arg-shades-line-alpha"></a></dt>
<dd><p>The transparency (alpha) of the lines around the rectangles (shades)</p></dd>


<dt id="arg-vline">vline<a class="anchor" aria-label="anchor" href="#arg-vline"></a></dt>
<dd><p>A number or vector of numbers to add a vertical line to the plot at a specific chromosomal position, e.g <code>vline="chr1:204000066"</code>. Multiple values can be provided in a vector, e.g  <code>vline=c("chr1:204000066","chr5:100500188")</code></p></dd>


<dt id="arg-vline-color">vline_color<a class="anchor" aria-label="anchor" href="#arg-vline-color"></a></dt>
<dd><p>A string. The color of added vertical line/s (default: grey)</p></dd>


<dt id="arg-vline-linetype">vline_linetype<a class="anchor" aria-label="anchor" href="#arg-vline-linetype"></a></dt>
<dd><p>A string. The linetype of added vertical line/s (default : dashed)</p></dd>


<dt id="arg-vline-alpha">vline_alpha<a class="anchor" aria-label="anchor" href="#arg-vline-alpha"></a></dt>
<dd><p>A number. The alpha of added vertical line/s (default : 1)</p></dd>


<dt id="arg-vline-size">vline_size<a class="anchor" aria-label="anchor" href="#arg-vline-size"></a></dt>
<dd><p>A number.The size of added vertical line/s (default : 0.5)</p></dd>


<dt id="arg-region">region<a class="anchor" aria-label="anchor" href="#arg-region"></a></dt>
<dd><p>A string representing a genetic region, e.g. chr1:67038906-67359979</p></dd>


<dt id="arg-theme-grey">theme_grey<a class="anchor" aria-label="anchor" href="#arg-theme-grey"></a></dt>
<dd><p>A logical scalar (default: FALSE). Use gray rectangles (instead of white to distinguish between chromosomes)</p></dd>


<dt id="arg-xaxis-label">xaxis_label<a class="anchor" aria-label="anchor" href="#arg-xaxis-label"></a></dt>
<dd><p>A string. The label for the x-axis (default: Chromosome)</p></dd>


<dt id="arg-use-shades">use_shades<a class="anchor" aria-label="anchor" href="#arg-use-shades"></a></dt>
<dd><p>A logical scalar (default: FALSE). Use shades/rectangles to distinguish between chromosomes</p></dd>


<dt id="arg-even-no-chr-lightness">even_no_chr_lightness<a class="anchor" aria-label="anchor" href="#arg-even-no-chr-lightness"></a></dt>
<dd><p>Lightness value for even numbered chromosomes. A number or vector of numbers between 0 and 1 (default: 0.8). If set to 0.5, the same color as shown for odd numbered chromosomes is displayed. A value below 0.5 will result in a darker color displayed for even numbered chromosomes, whereas a value above 0.5 results in a lighter color.</p></dd>


<dt id="arg-get-chr-lengths-from-data">get_chr_lengths_from_data<a class="anchor" aria-label="anchor" href="#arg-get-chr-lengths-from-data"></a></dt>
<dd><p>A logical scalar (default: TRUE). If set to FALSE, use the inbuilt chromosome lengths (from hg38), instead of chromosome lengths based on the max position for each chromosome in the input dataset/s.</p></dd>


<dt id="arg-log-trans-p">log_trans_p<a class="anchor" aria-label="anchor" href="#arg-log-trans-p"></a></dt>
<dd><p>A logical scalar (default: TRUE). By default the p-values in the input datasets are log transformed using -log10. Set this argument to FALSE if the p-values in the datasets have already been log transformed.</p></dd>


<dt id="arg-chr-ticknames">chr_ticknames<a class="anchor" aria-label="anchor" href="#arg-chr-ticknames"></a></dt>
<dd><p>A vector containing the chromosome names displayed on the x-axis. If NULL, the following format is used: chr_ticknames &lt;- c(1:16, ”,18, ”,20, ”,22, 'X')</p></dd>


<dt id="arg-show-all-chrticks">show_all_chrticks<a class="anchor" aria-label="anchor" href="#arg-show-all-chrticks"></a></dt>
<dd><p>A logical scalar (default : FALSE). Set to TRUE to show all the chromosome names on the ticks on the x-axis</p></dd>


<dt id="arg-hide-chrticks-from-pos">hide_chrticks_from_pos<a class="anchor" aria-label="anchor" href="#arg-hide-chrticks-from-pos"></a></dt>
<dd><p>A number (default: 17). Hide every nth chromosome name on the x-axis FROM this position (chromosome number)</p></dd>


<dt id="arg-hide-chrticks-to-pos">hide_chrticks_to_pos<a class="anchor" aria-label="anchor" href="#arg-hide-chrticks-to-pos"></a></dt>
<dd><p>A number (default: NULL). Hide every nth chromosome name on the x-axis TO this position (chromosome number). When NULL this variable will be set to the number of numeric chromosomes in the input dataset.</p></dd>


<dt id="arg-hide-every-nth-chrtick">hide_every_nth_chrtick<a class="anchor" aria-label="anchor" href="#arg-hide-every-nth-chrtick"></a></dt>
<dd><p>A number (default: 2). Hide every nth chromosome tick on the x-axis (from the hide_chr_ticks_from_pos to the hide_chr_ticks_to_pos).</p></dd>


<dt id="arg-downsample-cutoff">downsample_cutoff<a class="anchor" aria-label="anchor" href="#arg-downsample-cutoff"></a></dt>
<dd><p>A number (default: 0.05) used to downsample the input dataset prior to plotting. Sets the fraction of high p-value (default: P&gt;0.05) markers to display on the plot.</p></dd>


<dt id="arg-downsample-prop">downsample_prop<a class="anchor" aria-label="anchor" href="#arg-downsample-prop"></a></dt>
<dd><p>A number (default: 0.1) used to downsample the input dataset prior to plotting. Only a proportion of the variants (10% by default) with P-values higher than the downsample_cutoff will be displayed on the plot.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>ggplot object</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="fu">manhattan</span><span class="op">(</span><span class="va">CD_UKBB</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Thorhildur Juliusdottir, Andri Stefansson.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

