<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="manhattan() displays association results for the entire genome on a Manhattan plot.
Required parameter is at least one dataset (dataframe) containing the association data (with columns CHROM,POS,P in upper or lowercase)
All other input parameters are optional"><title>Create a Manhattan plot — manhattan • topr</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Create a Manhattan plot — manhattan"><meta property="og:description" content="manhattan() displays association results for the entire genome on a Manhattan plot.
Required parameter is at least one dataset (dataframe) containing the association data (with columns CHROM,POS,P in upper or lowercase)
All other input parameters are optional"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">topr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/topr.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/manhattan.html">Intro into manhattan</a>
    <a class="dropdown-item" href="../articles/manhattanExtra.html">Extending manhattan plots with manhattanExtra</a>
    <a class="dropdown-item" href="../articles/regionplot.html">Intro into regionplot</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/totajuliusd/topr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Create a Manhattan plot</h1>
      <small class="dont-index">Source: <a href="https://github.com/totajuliusd/topr/blob/HEAD/R/manhattan.R" class="external-link"><code>R/manhattan.R</code></a></small>
      <div class="d-none name"><code>manhattan.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>manhattan()</code> displays association results for the entire genome on a Manhattan plot.
Required parameter is at least one dataset (dataframe) containing the association data (with columns <code>CHROM,POS,P</code> in upper or lowercase)</p>
<p>All other input parameters are optional</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">manhattan</span><span class="op">(</span></span>
<span>  <span class="va">df</span>,</span>
<span>  ntop <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  title <span class="op">=</span> <span class="st">""</span>,</span>
<span>  annotate <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  color <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sign_thresh <span class="op">=</span> <span class="fl">5e-08</span>,</span>
<span>  sign_thresh_color <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>  sign_thresh_label_size <span class="op">=</span> <span class="fl">3.5</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">3.5</span>,</span>
<span>  size <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  shape <span class="op">=</span> <span class="fl">19</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  highlight_genes_color <span class="op">=</span> <span class="st">"darkred"</span>,</span>
<span>  highlight_genes_ypos <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  axis_text_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  axis_title_size <span class="op">=</span> <span class="fl">14</span>,</span>
<span>  title_text_size <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  legend_title_size <span class="op">=</span> <span class="fl">13</span>,</span>
<span>  legend_text_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  protein_coding_only <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  angle <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  legend_labels <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  chr <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  annotate_with <span class="op">=</span> <span class="st">"Gene_Symbol"</span>,</span>
<span>  region_size <span class="op">=</span> <span class="fl">2e+07</span>,</span>
<span>  legend_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  legend_position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>  nudge_x <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  nudge_y <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>  xmin <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  xmax <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ymin <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ymax <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  highlight_genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  label_color <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  legend_nrow <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_label_size <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_label_angle <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  scale <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  show_legend <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sign_thresh_linetype <span class="op">=</span> <span class="st">"dashed"</span>,</span>
<span>  sign_thresh_size <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  rsids <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  rsids_color <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  rsids_with_vline <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  annotate_with_vline <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  shades_color <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  shades_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  segment.size <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  segment.color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>  segment.linetype <span class="op">=</span> <span class="st">"dashed"</span>,</span>
<span>  max.overlaps <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  label_fontface <span class="op">=</span> <span class="st">"plain"</span>,</span>
<span>  label_family <span class="op">=</span> <span class="st">""</span>,</span>
<span>  gene_label_fontface <span class="op">=</span> <span class="st">"plain"</span>,</span>
<span>  gene_label_family <span class="op">=</span> <span class="st">""</span>,</span>
<span>  build <span class="op">=</span> <span class="fl">38</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  label_alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  shades_line_alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  vline <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  vline_color <span class="op">=</span> <span class="st">"grey"</span>,</span>
<span>  vline_linetype <span class="op">=</span> <span class="st">"dashed"</span>,</span>
<span>  vline_alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  vline_size <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  region <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  theme_grey <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  xaxis_label <span class="op">=</span> <span class="st">"Chromosome"</span>,</span>
<span>  use_shades <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  even_no_chr_lightness <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  get_chr_lengths_from_data <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  log_trans_p <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  chr_ticknames <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  show_all_chrticks <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  hide_chrticks_from_pos <span class="op">=</span> <span class="fl">17</span>,</span>
<span>  hide_chrticks_to_pos <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  hide_every_nth_chrtick <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  downsample_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  downsample_prop <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>df</dt>
<dd><p>Dataframe or a list of dataframes (required columns are <code>CHROM,POS,P</code>), in upper- or lowercase) of association results.</p></dd>


<dt>ntop</dt>
<dd><p>An integer, number of datasets (GWAS results) to show on the top plot</p></dd>


<dt>title</dt>
<dd><p>A string to set the plot title</p></dd>


<dt>annotate</dt>
<dd><p>A number (p-value). Display annotation for variants with p-values below this threshold</p></dd>


<dt>color</dt>
<dd><p>A string or a vector of strings, for setting the color of the datapoints on the plot</p></dd>


<dt>sign_thresh</dt>
<dd><p>A number or vector of numbers, setting the horizontal significance threshold (default: <code>sign_thresh=5e-8</code>). Set to NULL to hide the significance threshold.</p></dd>


<dt>sign_thresh_color</dt>
<dd><p>A string or vector of strings to set the color/s of the significance threshold/s</p></dd>


<dt>sign_thresh_label_size</dt>
<dd><p>A number setting the text size of the label for the significance thresholds (default text size is 3.5)</p></dd>


<dt>label_size</dt>
<dd><p>An number to set the size of the plot labels (default: <code>label_size=3</code>)</p></dd>


<dt>size</dt>
<dd><p>A number or a vector of numbers, setting the size of the plot points (default: <code>size=1.2</code>)</p></dd>


<dt>shape</dt>
<dd><p>A number of a vector of numbers setting the shape of the plotted points</p></dd>


<dt>alpha</dt>
<dd><p>A number or a vector of numbers setting the transparency of the plotted points</p></dd>


<dt>highlight_genes_color</dt>
<dd><p>A string, color for the highlighted genes (default: darkred)</p></dd>


<dt>highlight_genes_ypos</dt>
<dd><p>An integer, controlling where on the y-axis the highlighted genes are placed (default value is 1)</p></dd>


<dt>axis_text_size</dt>
<dd><p>A number, size of the x and y axes tick labels (default: 12)</p></dd>


<dt>axis_title_size</dt>
<dd><p>A number, size of the x and y title labels (default: 12)</p></dd>


<dt>title_text_size</dt>
<dd><p>A number, size of the plot title (default: 13)</p></dd>


<dt>legend_title_size</dt>
<dd><p>A number, size of the legend title</p></dd>


<dt>legend_text_size</dt>
<dd><p>A number, size of the legend text</p></dd>


<dt>protein_coding_only</dt>
<dd><p>A logical scalar, if TRUE, only protein coding genes are used for annotation</p></dd>


<dt>angle</dt>
<dd><p>A number, the angle of the text label</p></dd>


<dt>legend_labels</dt>
<dd><p>A string or vector of strings representing legend labels for the input datasets</p></dd>


<dt>chr</dt>
<dd><p>A string or integer, the chromosome to plot (i.e. chr15), only required if the input dataframe contains results from more than one chromosome</p></dd>


<dt>annotate_with</dt>
<dd><p>A string. Annotate the variants with either Gene_Symbol or ID (default: "Gene_Symbol")</p></dd>


<dt>region_size</dt>
<dd><p>An integer (default = 20000000) (or a string represented as 200kb or 2MB) indicating the window size for variant labeling. Increase this number for sparser annotation and decrease for denser annotation.</p></dd>


<dt>legend_name</dt>
<dd><p>A string, use to change the name of the legend (default: None)</p></dd>


<dt>legend_position</dt>
<dd><p>A string, top,bottom,left or right</p></dd>


<dt>nudge_x</dt>
<dd><p>A number to vertically adjust the starting position of each gene label (this is a ggrepel parameter)</p></dd>


<dt>nudge_y</dt>
<dd><p>A number to horizontally adjust the starting position of each gene label (this is a ggrepel parameter)</p></dd>


<dt>xmin, xmax</dt>
<dd><p>Integer, setting the chromosomal range to display on the x-axis</p></dd>


<dt>ymin, ymax</dt>
<dd><p>Integer, min and max of the y-axis, (default values: <code>ymin=0, ymax=max(-log10(df$P))</code>)</p></dd>


<dt>highlight_genes</dt>
<dd><p>A string or vector of strings, gene or genes to highlight at the bottom of the plot</p></dd>


<dt>label_color</dt>
<dd><p>A string or a vector of strings. To change the color of the gene or variant labels</p></dd>


<dt>legend_nrow</dt>
<dd><p>An integer, sets the number of rows allowed for the legend labels</p></dd>


<dt>gene_label_size</dt>
<dd><p>A number setting the size of the gene labels shown at the bottom of the plot</p></dd>


<dt>gene_label_angle</dt>
<dd><p>A number setting the angle of the gene label shown at the bottom of the plot (default: 0)</p></dd>


<dt>scale</dt>
<dd><p>A number, to change the size of the title and axes labels and ticks at the same time (default : 1)</p></dd>


<dt>show_legend</dt>
<dd><p>A logical scalar, set to FALSE to hide the legend (default : TRUE)</p></dd>


<dt>sign_thresh_linetype</dt>
<dd><p>A string, the line-type of the horizontal significance threshold (default : dashed)</p></dd>


<dt>sign_thresh_size</dt>
<dd><p>A number, sets the size of the horizontal significance threshold line (default : 1)</p></dd>


<dt>rsids</dt>
<dd><p>A string (rsid) or vector of strings to highlight on the plot, e.g. <code>rsids=c("rs1234, rs45898")</code></p></dd>


<dt>rsids_color</dt>
<dd><p>A string, the color of the variants in variants_id (default color is red)</p></dd>


<dt>rsids_with_vline</dt>
<dd><p>A string (rsid) or vector of strings to highlight on the plot with their rsids and vertical lines further highlighting their positions</p></dd>


<dt>annotate_with_vline</dt>
<dd><p>A number (p-value). Display annotation and vertical lines for variants with p-values below this threshold</p></dd>


<dt>shades_color</dt>
<dd><p>The color of the rectangles (shades) representing the different chromosomes on the Manhattan plot</p></dd>


<dt>shades_alpha</dt>
<dd><p>The transparency (alpha) of the rectangles (shades)</p></dd>


<dt>segment.size</dt>
<dd><p>line segment color (ggrepel argument)</p></dd>


<dt>segment.color</dt>
<dd><p>line segment thickness (ggrepel argument)</p></dd>


<dt>segment.linetype</dt>
<dd><p>line segment solid, dashed, etc.(ggrepel argument)</p></dd>


<dt>max.overlaps</dt>
<dd><p>Exclude text labels that overlap too many things. Defaults to 10 (ggrepel argument)</p></dd>


<dt>label_fontface</dt>
<dd><p>A string or a vector of strings. Label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p></dd>


<dt>label_family</dt>
<dd><p>A string or a vector of strings. Label font name (default ggrepel argument is "")</p></dd>


<dt>gene_label_fontface</dt>
<dd><p>Gene label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p></dd>


<dt>gene_label_family</dt>
<dd><p>Gene label font name (default ggrepel argument is "")</p></dd>


<dt>build</dt>
<dd><p>A number representing the genome build or a data frame. Set to 37 to change to build (GRCh37). The default is build 38 (GRCh38).</p></dd>


<dt>verbose</dt>
<dd><p>A logical scalar (default: NULL). Set to FALSE to suppress printed messages</p></dd>


<dt>label_alpha</dt>
<dd><p>An number or vector of numbers to set the transparency of the plot labels (default: <code>label_alpha=1</code>)</p></dd>


<dt>shades_line_alpha</dt>
<dd><p>The transparency (alpha) of the lines around the rectangles (shades)</p></dd>


<dt>vline</dt>
<dd><p>A number or vector of numbers to add a vertical line to the plot at a specific chromosomal position, e.g <code>vline="chr1:204000066"</code>. Multiple values can be provided in a vector, e.g  <code>vline=c("chr1:204000066","chr5:100500188")</code></p></dd>


<dt>vline_color</dt>
<dd><p>A string. The color of added vertical line/s (default: grey)</p></dd>


<dt>vline_linetype</dt>
<dd><p>A string. The linetype of added vertical line/s (default : dashed)</p></dd>


<dt>vline_alpha</dt>
<dd><p>A number. The alpha of added vertical line/s (default : 1)</p></dd>


<dt>vline_size</dt>
<dd><p>A number.The size of added vertical line/s (default : 0.5)</p></dd>


<dt>region</dt>
<dd><p>A string representing a genetic region, e.g. chr1:67038906-67359979</p></dd>


<dt>theme_grey</dt>
<dd><p>A logical scalar (default: FALSE). Use gray rectangles (instead of white to distinguish between chromosomes)</p></dd>


<dt>xaxis_label</dt>
<dd><p>A string. The label for the x-axis (default: Chromosome)</p></dd>


<dt>use_shades</dt>
<dd><p>A logical scalar (default: FALSE). Use shades/rectangles to distinguish between chromosomes</p></dd>


<dt>even_no_chr_lightness</dt>
<dd><p>Lightness value for even numbered chromosomes. A number or vector of numbers between 0 and 1 (default: 0.8). If set to 0.5, the same color as shown for odd numbered chromosomes is displayed. A value below 0.5 will result in a darker color displayed for even numbered chromosomes, whereas a value above 0.5 results in a lighter color.</p></dd>


<dt>get_chr_lengths_from_data</dt>
<dd><p>A logical scalar (default: TRUE). If set to FALSE, use the inbuilt chromosome lengths (from hg38), instead of chromosome lengths based on the max position for each chromosome in the input dataset/s.</p></dd>


<dt>log_trans_p</dt>
<dd><p>A logical scalar (default: TRUE). By default the p-values in the input datasets are log transformed using -log10. Set this argument to FALSE if the p-values in the datasets have already been log transformed.</p></dd>


<dt>chr_ticknames</dt>
<dd><p>A vector containing the chromosome names displayed on the x-axis. If NULL, the following format is used: chr_ticknames &lt;- c(1:16, '',18, '',20, '',22, 'X')</p></dd>


<dt>show_all_chrticks</dt>
<dd><p>A logical scalar (default : FALSE). Set to TRUE to show all the chromosome names on the ticks on the x-axis</p></dd>


<dt>hide_chrticks_from_pos</dt>
<dd><p>A number (default: 17). Hide every nth chromosome name on the x-axis FROM this position (chromosome number)</p></dd>


<dt>hide_chrticks_to_pos</dt>
<dd><p>A number (default: NULL). Hide every nth chromosome name on the x-axis TO this position (chromosome number). When NULL this variable will be set to the number of numeric chromosomes in the input dataset.</p></dd>


<dt>hide_every_nth_chrtick</dt>
<dd><p>A number (default: 2). Hide every nth chromosome tick on the x-axis (from the hide_chr_ticks_from_pos to the hide_chr_ticks_to_pos).</p></dd>


<dt>downsample_cutoff</dt>
<dd><p>A number (default: 0.05) used to downsample the input dataset prior to plotting. Sets the fraction of high p-value (default: P&gt;0.05) markers to display on the plot.</p></dd>


<dt>downsample_prop</dt>
<dd><p>A number (default: 0.1) used to downsample the input dataset prior to plotting. Only a proportion of the variants (10% by default) with P-values higher than the downsample_cutoff will be displayed on the plot.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>ggplot object</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="fu">manhattan</span><span class="op">(</span><span class="va">CD_UKBB</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Thorhildur Juliusdottir, Andri Stefansson.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

