<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting started with topr • topr</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting started with topr">
<meta name="description" content="The topr package is a set of functions used to visualize, explore and annotate genetic association results.
">
<meta property="og:description" content="The topr package is a set of functions used to visualize, explore and annotate genetic association results.
">
<meta property="og:image" content="https://kscott-1.github.io/topr/logo.svg">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">topr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/topr.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/locuszoom.html">Locuszoom plot examples</a></li>
    <li><a class="dropdown-item" href="../articles/manhattanExtra.html">Extending manhattan plots with manhattanExtra</a></li>
    <li><a class="dropdown-item" href="../articles/moreManhattan.html">More manhattan examples</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/totajuliusd/topr/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Getting started with topr</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/totajuliusd/topr/blob/doc_testing/vignettes/topr.Rmd" class="external-link"><code>vignettes/topr.Rmd</code></a></small>
      <div class="d-none name"><code>topr.Rmd</code></div>
    </div>

    
    
<p>The topr package is a set of functions used to visualize, explore and
annotate genetic association results. In particular, using only the
chromosome, position and p-value of each variant in results dataframes,
topr makes it easy to display gene annotations and multiple traits in a
single manhattan plot. Other functionality includes regional plots &amp;
locus zoom plots.</p>
<div class="section level2">
<h2 id="exploration-with-built-in-data">Exploration with built-in data<a class="anchor" aria-label="anchor" href="#exploration-with-built-in-data"></a>
</h2>
<p>The package comes with 3 sample datasets, namely
<code>CD_UKBB</code>, <code>CD_FINNGEN</code> &amp;
<code>UC_UKBB</code>.</p>
<p>Let’s look at the <code>CD_UKBB</code> dataset:</p>
<pre><code><span><span class="co">#&gt;   CHROM     POS          ID       REF ALT           P       OR      AF</span></span>
<span><span class="co">#&gt; 1  chr1 1006415 rs145588482 TGGCAGCTC   T 0.000468758 0.583384 0.01293</span></span>
<span><span class="co">#&gt; 2  chr1 1007256  rs76233940         G   A 0.000401567 0.579783 0.01301</span></span>
<span><span class="co">#&gt; 3  chr1 1341559 rs376494450         C   T 0.000151216 1.320130 0.02706</span></span>
<span><span class="co">#&gt; 4  chr1 1480224  rs71628972         A   G 0.000760160 1.412750 0.01319</span></span>
<span><span class="co">#&gt; 5  chr1 2035619 rs111334586         C   T 0.000920615 1.413370 0.01209</span></span>
<span><span class="co">#&gt; 6  chr1 2213349    rs262695         A   G 0.000881453 0.904873 0.32073</span></span></code></pre>
<p>Note that while there are extra columns in the sample data, the only
columns required for plotting are CHROM, POS and P.</p>
<div class="section level3">
<h3 id="single-dataset-plotting">Single dataset plotting<a class="anchor" aria-label="anchor" href="#single-dataset-plotting"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="va">CD_UKBB</span><span class="op">)</span></span></code></pre></div>
<p><img src="topr_files/figure-html/ex-manh-1.png" width="1152"></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, annotate<span class="op">=</span><span class="fl">5e-9</span><span class="op">)</span></span></code></pre></div>
<p><img src="topr_files/figure-html/ex-manh2-1.png" width="1152"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/regionplot.html">regionplot</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, gene<span class="op">=</span><span class="st">"IL23R"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Zoomed to region:  1:67038907-67359979"</span></span></code></pre></div>
<p><img src="topr_files/figure-html/ex-region-1.png" width="1152"></p>
</div>
<div class="section level3">
<h3 id="multiple-dataset-plotting">Multiple dataset plotting<a class="anchor" aria-label="anchor" href="#multiple-dataset-plotting"></a>
</h3>
<p>In addition to plotting a single dataset, topr can also plot multiple
datasets in a single manhattan/region plot.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span><span class="op">)</span>, legend_labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UKBB"</span>, <span class="st">"FinnGen"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="topr_files/figure-html/ex-multimanh-1.png" width="1152"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/regionplot.html">regionplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span><span class="op">)</span>, gene<span class="op">=</span><span class="st">"IL23R"</span>, legend_labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UKBB"</span>, <span class="st">"FinnGen"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Zoomed to region:  1:67038907-67359979"</span></span></code></pre></div>
<p><img src="topr_files/figure-html/ex-multiregion-1.png" width="1152"></p>
</div>
<div class="section level3">
<h3 id="other-useful-functions">Other useful functions<a class="anchor" aria-label="anchor" href="#other-useful-functions"></a>
</h3>
<p>Extract lead/index variants from the GWAS dataset (CD_UKBB):</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_lead_snps.html">get_lead_snps</a></span><span class="op">(</span><span class="va">CD_UKBB</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">P</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;     CHROM      POS          ID REF ALT           P       OR      AF</span></span>
<span><span class="co">#&gt; 6.1  chr6 31660620 rs148844907   T   A 1.47082e-24 2.433450 0.01034</span></span>
<span><span class="co">#&gt; 6.2  chr6 32205822 rs138753323   T   C 7.13519e-24 1.844850 0.02882</span></span>
<span><span class="co">#&gt; 16  chr16 50729867   rs2066847   G  GC 7.36933e-24 2.142320 0.01588</span></span>
<span><span class="co">#&gt; 1.1  chr1 67216513  rs11576518   G   A 8.03684e-20 0.777437 0.44081</span></span>
<span><span class="co">#&gt; 6.3  chr6 32708532 rs144614916   A   C 1.27821e-15 1.801210 0.01997</span></span>
<span><span class="co">#&gt; 7    chr7 50274703   rs2219345   T   G 8.52335e-14 1.230940 0.59313</span></span></code></pre></div>
<p>Annotate the lead/index variants with their nearest gene &amp;
biotype:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_lead_snps.html">get_lead_snps</a></span><span class="op">(</span><span class="va">CD_UKBB</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/annotate_with_nearest_gene.html">annotate_with_nearest_gene</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">P</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;     CHROM      POS          ID REF ALT           P       OR      AF Gene_Symbol</span></span>
<span><span class="co">#&gt; 6.1  chr6 31660620 rs148844907   T   A 1.47082e-24 2.433450 0.01034     C6orf47</span></span>
<span><span class="co">#&gt; 6.2  chr6 32205822 rs138753323   T   C 7.13519e-24 1.844850 0.02882      NOTCH4</span></span>
<span><span class="co">#&gt; 16  chr16 50729867   rs2066847   G  GC 7.36933e-24 2.142320 0.01588        NOD2</span></span>
<span><span class="co">#&gt; 1.1  chr1 67216513  rs11576518   G   A 8.03684e-20 0.777437 0.44081    C1orf141</span></span>
<span><span class="co">#&gt; 6.3  chr6 32708532 rs144614916   A   C 1.27821e-15 1.801210 0.01997     MTCO3P1</span></span>
<span><span class="co">#&gt; 7    chr7 50274703   rs2219345   T   G 8.52335e-14 1.230940 0.59313       IKZF1</span></span>
<span><span class="co">#&gt;                    biotype</span></span>
<span><span class="co">#&gt; 6.1         protein_coding</span></span>
<span><span class="co">#&gt; 6.2         protein_coding</span></span>
<span><span class="co">#&gt; 16          protein_coding</span></span>
<span><span class="co">#&gt; 1.1         protein_coding</span></span>
<span><span class="co">#&gt; 6.3 unprocessed_pseudogene</span></span>
<span><span class="co">#&gt; 7           protein_coding</span></span></code></pre></div>
<p>Get genomic coordinates for a gene (topr uses genome build GRCh38.p13
by default):</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_gene_coords.html">get_gene_coords</a></span><span class="op">(</span><span class="st">"IL23R"</span><span class="op">)</span></span>
<span><span class="co">#&gt;   chrom gene_start gene_end gene_symbol        biotype</span></span>
<span><span class="co">#&gt; 1     1   67138907 67259979       IL23R protein_coding</span></span></code></pre></div>
<p>Get genomic coordinates for a gene using genome build GRCh37
instead.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_gene_coords.html">get_gene_coords</a></span><span class="op">(</span><span class="st">"IL23R"</span>, build<span class="op">=</span><span class="st">"37"</span><span class="op">)</span></span>
<span><span class="co">#&gt;   chrom gene_start gene_end gene_symbol        biotype</span></span>
<span><span class="co">#&gt; 1     1   67632083 67725662       IL23R protein_coding</span></span></code></pre></div>
<p>Get snps within a region:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_snps_within_region.html">get_snps_within_region</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, region<span class="op">=</span><span class="st">"chr1:67138906-67259979"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;   CHROM      POS         ID REF ALT           P       OR      AF</span></span>
<span><span class="co">#&gt; 1  chr1 67139451 rs10789224   T   C 3.11449e-09 0.837104 0.31415</span></span>
<span><span class="co">#&gt; 2  chr1 67140456  rs1158199   T   C 3.65767e-04 1.101210 0.53528</span></span>
<span><span class="co">#&gt; 3  chr1 67141399  rs9729046   C   G 3.91140e-06 0.875025 0.34477</span></span>
<span><span class="co">#&gt; 4  chr1 67141452   rs736197   G   T 3.91082e-06 0.875021 0.34477</span></span>
<span><span class="co">#&gt; 5  chr1 67142417 rs11582178   C   T 5.10929e-10 0.613183 0.04768</span></span>
<span><span class="co">#&gt; 6  chr1 67143664   rs963427   A   G 3.36454e-06 0.874194 0.34362</span></span></code></pre></div>
<p>Get the top variant on a chromosome:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_top_snp.html">get_top_snp</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, chr<span class="op">=</span><span class="st">"chr1"</span><span class="op">)</span></span>
<span><span class="co">#&gt;   CHROM      POS         ID REF ALT           P       OR      AF</span></span>
<span><span class="co">#&gt; 1  chr1 67216513 rs11576518   G   A 8.03684e-20 0.777437 0.44081</span></span></code></pre></div>
<p>Create a snpset by extracting the top/lead variants from the CD_UKBB
dataset and overlapping variants (by position) in the CD_FINNGEN
dataset.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_snpset.html">get_snpset</a></span><span class="op">(</span><span class="va">CD_UKBB</span>, <span class="va">CD_FINNGEN</span><span class="op">)</span></span>
<span><span class="co">#&gt; $matched</span></span>
<span><span class="co">#&gt;   CHROM       POS REF1 ALT1          P1        E1 REF2 ALT2          P2</span></span>
<span><span class="co">#&gt; 1     1 206770623    C    A 7.85424e-09 0.2021732    C    A 1.22098e-04</span></span>
<span><span class="co">#&gt; 2    16  50729867    G   GC 7.36933e-24 0.7618894    G   GC 1.49882e-09</span></span>
<span><span class="co">#&gt; 3     5  40439961    C    T 7.42668e-11 0.1818381    C    T 2.80350e-13</span></span>
<span><span class="co">#&gt; 4     5 159399784    C    A 7.12812e-09 0.1680029    C    A 2.37432e-05</span></span>
<span><span class="co">#&gt; 5     1  67216513    A    G 8.03684e-20 0.2517527    A    G 1.04321e-08</span></span>
<span><span class="co">#&gt;         E2         ID Gene_Symbol        biotype</span></span>
<span><span class="co">#&gt; 1 0.130601  rs3024493        IL10 protein_coding</span></span>
<span><span class="co">#&gt; 2 0.530875  rs2066847        NOD2 protein_coding</span></span>
<span><span class="co">#&gt; 3 0.187984  rs7713270       TTC33 protein_coding</span></span>
<span><span class="co">#&gt; 4 0.114921  rs6871626       IL12B protein_coding</span></span>
<span><span class="co">#&gt; 5 0.146556 rs11576518    C1orf141 protein_coding</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $snp_not_found_in_df2</span></span>
<span><span class="co">#&gt;     CHROM       POS REF1 ALT1          P1         E1      ID_tmp</span></span>
<span><span class="co">#&gt; 2.2     2  39590816    G    C 2.13296e-08  0.1589079  2_39590816</span></span>
<span><span class="co">#&gt; 2.1     2 233237298    A    C 1.70216e-09  0.1628839 2_233237298</span></span>
<span><span class="co">#&gt; 6.1     6  31660620    T    A 1.47082e-24  0.8893100  6_31660620</span></span>
<span><span class="co">#&gt; 6.2     6  32205822    T    C 7.13519e-24  0.6123980  6_32205822</span></span>
<span><span class="co">#&gt; 6.3     6  32708532    A    C 1.27821e-15  0.5884587  6_32708532</span></span>
<span><span class="co">#&gt; 7       7  50274703    T    G 8.52335e-14  0.2077781  7_50274703</span></span>
<span><span class="co">#&gt; 9       9   4984530    G    C 5.03745e-11 -0.1833112   9_4984530</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $snp_found_different_alleles_in_df2</span></span>
<span><span class="co">#&gt;  [1] CHROM  POS    REF1   ALT1   P1     E1     REF2   ALT2   P2     E2    </span></span>
<span><span class="co">#&gt; [11] ID     ID_tmp</span></span>
<span><span class="co">#&gt; &lt;0 rows&gt; (or 0-length row.names)</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Thorhildur Juliusdottir, Andri Stefansson.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
